stl/inc/iterator

```diff
@@ -919,8 +923,14 @@ public:
             common_iterator _Tmp = *this;
             ++_Val._Iterator;
             return _Tmp;
-        } else {
+        } else if constexpr (_Can_reference<_Iter> //

```

Comment:
`_Can_reference<_Iter>` is always `true` since `_Iter` is an object type (`_Can_reference<T>` is true iff `T&` is a valid type). This is not checking what we need to check. Given that we need a requires-expression, we should probably just extract this entire `if constexpr` condition out into a named concept.

Reply:
I pulled the `_Can_reference` part out into a named concept but left the other stuff here, as the concept would get super clunky